test/dynamics/pulse/test_string_model_parser.py

```diff
@@ -0,0 +1,422 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+"""
+Tests for pulse.string_model_parser for importing qiskit.pulse model string representation.
+"""
+
+import numpy as np
+
+from qiskit import QiskitError
+from qiskit.quantum_info.operators import Operator
+
+from qiskit_dynamics.pulse import parse_hamiltonian_dict
+from qiskit_dynamics.pulse.string_model_parser.string_model_parser import (
+    hamiltonian_pre_parse_exceptions,
+)
+
+from qiskit_dynamics.type_utils import to_array
+
+from ..common import QiskitDynamicsTestCase
+
+
+class TestHamiltonianPreParseExceptions(QiskitDynamicsTestCase):
+    """Tests for preparse formatting exceptions."""
+
+    def test_no_h_str(self):
```

Comment:
Super unimportant but just to cover edge case of `h_str= ""`, add a line for that as well, since we have the code for it above. (Let me know if this is too unimportant, still trying to figure out best practices here)

Reply:
Added two additional checks to this test to verify that the same error gets raised if the argument is: `{'h_str': []}` or `{'h_str': ['']}`.

This is good to point - obviously we can't be overly exhaustive of all possible validation issues but it's good to make sure nothing breaks here.